Sources/Fisticuffs/SubscriptionCollection.swift

```diff
@@ -37,7 +37,9 @@ open class SubscriptionCollection<T> {
     public func notify(time: NotifyWhen, old: T?, new: T) {
         lock.withLock {
             for s in subscriptions where s.when == time {
-                s.callback(old, new)
+                s.scheduler.schedule {
+                    s.callback(old, new)
+                }
             }
```

Comment:
Not super important but what do you think on updating `s` to `subscription` instead?

Reply:
Sure, updated